### PR TITLE
Remove TransactionType

### PR DESCRIPTION
**Pull request by adagys**
Now that have separate classes for notary change transactions, we can remove the TransactionType concept.

It's a lot of files, but the changes are simple removals/replacements and I grouped it in commits.

Note to self: need to update Cordapps before merging

### CODE REVIEW

File: docs/source/api-transactions.rst [hunk 1]
```diff
@@ -590,3 +560,22 @@ Notarising and recording
...
 Notarising and recording a transaction is handled by a built-in flow called ``FinalityFlow``. See
 :doc:`api-flows` for more details.
+
+Notary-change transactions
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+A single Corda network will usually have multiple notary services. To commit a transaction, we require a signature
+from the notary service associated with each input state. If we tried to commit a transaction where the input
+states were associated with different notary services, the transaction would require a signature from multiple notary
+services, creating a complicated multi-phase commit scenario. To prevent this, every input state in a transaction
+must be associated the same notary.
```
Comment (shamsasari) [inline docs/source/api-transactions.rst]: "associated with"

Comment (joeldudleyr3) [inline docs/source/api-transactions.rst]: Haha, my bad.

File: docs/source/changelog.rst [hunk 2]
```diff
@@ -856,7 +856,7 @@ Summary of API changes (not exhaustive):
       Currently two types are supported: General (runs the default build and validation logic) and NotaryChange (
       contract code is not run during validation, checks that the notary field is the only difference between the
       inputs and outputs).
-      ``TransactionBuilder()`` is now abstract, you should use ``TransactionType.General.Builder()`` for building transactions.
+      ``TransactionBuilder()`` is now abstract, you should use ``TransactionBuilder()`` for building transactions.
```
Comment (joeldudleyr3) [inline docs/source/changelog.rst]: Not sure we should be changing the change log?

**Comment by joeldudleyr3** [comment]
Spoke to Andrius - LGTM, apart from the typo and the change to the changelog.

File: docs/source/changelog.rst (hunk 2 above)
Comment (adagys) [inline docs/source/changelog.rst]: Whoops, good catch

**Comment by adagys** [comment]
This PR depends on https://github.com/corda/corda/pull/1125

**Comment by adagys** [comment]
Yeah, but I only noticed that now. There's an issue with a test that will happen to go away once that other PR is merged.

**Pull request merged**